scripts/gen-i18n.js

```diff
@@ -73,6 +74,41 @@ function getTKey(arg) {
     return null;
 }
 
+function getFormatStrings(str) {
+    // Match anything that starts with %
+    // We could make a regex that matched the full placeholder, but this
+    // would just not match invalid placeholders and so wouldn't help us
+    // detect the invalid ones.
+    // Also note that for simplicity, this just matches a % character and then
+    // anything up to the next % character (or a single %, or end of string).
+    const formatStringRe = /%([^%]+|%|$)/g;
+    const formatStrings = new Set();
+
+    let match;
+    while ( (match = formatStringRe.exec(str)) !== null) {
+        const placeholder = match[1]; // Minus the leading '%'
+        if (placeholder === '%') continue; // Literal % is %%
+
+        const placeholderMatch = placeholder.match(/^\((.*?)\)(.)/);
+        if (placeholderMatch === null) {
+            throw new Error("Invalid format specifier: '"+match[0]+"'");
+        }
+        if (placeholderMatch.length < 3) {
+            throw new Error("Malformed format specifier");
+        }
+        const placeHolderName = placeholderMatch[1];
+        const placeHolderFormat = placeholderMatch[2];
+
+        if (placeHolderFormat !== 's') {
+            throw new Error(`'${placeHolderFormat}' used as format character: you probably didn't mean this`);
```

Comment:
Maybe mention that only "s" is supported? as much as I would love a "computer says 'no'"-style response :wink:

Reply:
heh, yes, fair